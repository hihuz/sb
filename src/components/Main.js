import React from "react";
import Intro from "./Intro";
import Section from "./Section";

const Main = ({ intro, articles }) => (
  <main>
    <Intro {...intro} />
    {articles.map(article => <Section key={article.name} {...article} />)}
  </main>
);

export default Main;
