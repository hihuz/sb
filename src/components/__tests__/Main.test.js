import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Main from "../Main";

describe("Main", () => {
  test("Snapshot", () => {
    const component = shallow(<Main articles={[]} />);
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });

  test("Should render an Intro component and pass it its intro prop", () => {
    const intro = {
      title: "foo",
      text: "bar",
    };
    const component = shallow(<Main intro={intro} articles={[]} />);
    const child = component.find("Intro");
    expect(child.length).toEqual(1);
    expect(child.props().title).toEqual("foo");
    expect(child.props().text).toEqual("bar");
  });

  test("Should render a number of Sections components based off article prop", () => {
    const articles = [
      { name: "foo", title: "bla" },
      { name: "bar", title: "bleh" },
      { name: "baz", title: "bluh" },
    ];
    const component = shallow(<Main articles={articles} />);
    const sections = component.find("Section");
    expect(sections.length).toEqual(3);
    articles.forEach(({ name, title }, i) => {
      const child = sections.at(i);
      expect(child.props().title).toEqual(title);
      expect(child.props().name).toEqual(name);
    });
  });
});
