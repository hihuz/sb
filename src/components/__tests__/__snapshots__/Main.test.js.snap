// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Snapshot 1`] = `
<main>
  <Intro />
</main>
`;
